File: src/main/java/com/pboucher/mailSender/controller/MailController.java
package com.pboucher.mailSender.controller;

import com.pboucher.mailSender.metier.Contact;
import com.pboucher.mailSender.service.MailService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import java.util.HashMap;
import java.util.List;

@RestController
public class MailController {

    @Resource
    private MailService mailService;

    @RequestMapping(value="/contacts/{montant}", method = RequestMethod.POST)
    @ResponseBody
    public ResponseEntity contacts(@PathVariable("montant") int montant, @RequestBody List<Contact> contacts) {
        HashMap<Contact, Contact> shuffleContacts = mailService.shuffleContacts(contacts);
        mailService.sendSimpleMail(shuffleContacts, montant);
        return ResponseEntity.ok(contacts);
    }
}

File: src/main/kotlin/com/pboucher/mailsender/controller/MailController.kt
package com.pboucher.mailsender.controller

import com.pboucher.mailsender.metier.Contact
import com.pboucher.mailsender.service.MailService
import com.pboucher.mailsender.service.impl.MailServiceImpl
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import javax.annotation.Resource


@RestController
class MailController {

    @Resource
    private val mailService: MailService = MailServiceImpl()

    @RequestMapping(value = ["/contacts/{montant}"], method = [RequestMethod.POST])
    @ResponseBody
    fun contacts(@PathVariable("montant") montant: Int, @RequestBody contacts: List<Contact>): ResponseEntity<*>? {
        val shuffleContacts: HashMap<Contact, Contact> = mailService.shuffleContacts(contacts)
        mailService.sendSimpleMail(shuffleContacts, montant)
        return ResponseEntity.ok(contacts)
    }

}
